pkg/binding/binding-controller.go

```diff
@@ -601,6 +603,29 @@ func (c *Controller) setupBindingInformer(ctx context.Context) error {
 	return nil
 }
 
+func shouldQueueBindingOnUpdate(old, new *v1alpha1.Binding) bool {
+	if old.Generation != new.Generation {
+		return true
+	}
+	if len(old.Status.Conditions) != len(new.Status.Conditions) || len(old.Status.Errors) != len(new.Status.Errors) {
+		return true
+	}
+
+	// below are 'loose' element-by-element comparisons, without trying to recognize that reordering is insignificant
+	for i, c := range old.Status.Conditions {
+		if !v1alpha1.AreConditionsEqual(c, new.Status.Conditions[i]) {
+			return true
+		}
+	}
+	for i, e := range old.Status.Errors {
+		if e != new.Status.Errors[i] {
+			return true
+		}
+	}
```

Comment:
`abstract.SliceEqual` will handle cases where golang `==` is the right equality test.

Reply:
Also, I think that it would be good to enhance `abstract` with an alternative form of `SliceEqual` that takes a comparison function. I would curry it too, so that we can have an algebra of comparison functions.

```golang
func SliceEqualFunction[Elt comparable](equal func (Elt, Elt) bool) func (slice1, slice2 []Elt) bool {...}
```

Reply:
Used `abstract.SliceEqual`, thanks!

Reply:
Just saw the 2nd comment. Would that be overkill here? Or are you saying we contribute to Go?

Reply:
I am not proposing a contribution to golang here.